os/user/getent_cgo.go

```diff
@@ -0,0 +1,74 @@
+// +build aix dragonfly freebsd !android,linux netbsd openbsd solaris darwin
+// +build cgo,!osusergo
+
+package user
+
+/*
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+#include <grp.h>
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"sync"
+)
+
+// Since the getpwent is not thread safe, muUsers locks access to iterateUsers for single goroutine at a time.
```

Comment:
A mutex is a considerable price to pay for every iteration. The callers should either not call it concurrently, or have their own locking. That, of course, needs to be documented.